conan/tools/microsoft/msbuilddeps.py

```diff
@@ -126,10 +127,13 @@ def _condition(self):
     @staticmethod
     def _dep_name(dep, build):
         dep_name = dep.ref.name
-        dep_name = dep_name.replace(".", "_")
         if build:  # dep.context == CONTEXT_BUILD:
             dep_name += "_build"
-        return dep_name
+        return MSBuildDeps._get_valid_xml_format(dep_name)
+
+    @staticmethod
+    def _get_valid_xml_format(name):
+        return EXCLUDED_SYMBOLS_PATTERN.sub("_", name)
```

Comment:
Maybe a ``x`` looks better in the generated files? That would leave the question is how to replace the ``-`` and the ``.`` too, so I guess it is good as it is.

Reply:
Yeah, I think it's good as it is, but it could be ok to make different replacements for each one. For instance:

* `.` -> `_`
* `+` or `-` -> `x`